packages/framework/src/types/validator.types.ts

```diff
@@ -19,15 +21,15 @@ export type ValidateResult<T> =
       data: T;
     };
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export interface Validator<T_Schema extends Schema = Schema> {
+export type Validator<T_Schema extends Schema = Schema> = {
   validate: <
     T_Unvalidated extends Record<string, unknown> = FromSchemaUnvalidated<T_Schema>,
     T_Validated extends Record<string, unknown> = FromSchema<T_Schema>,
   >(
     data: T_Unvalidated,
     schema: T_Schema
   ) => Promise<ValidateResult<T_Validated>>;
-  canHandle: (schema: Schema) => schema is T_Schema;
+  canHandle: (schema: Schema) => Promise<boolean>;
```

Comment:
We have to forgo the type guard here unfortunately, because Typescript doesn't support async type guards yet :/